src/cluster_legacy.c

```diff
@@ -3989,28 +4146,64 @@ int clusterSendModuleMessageToTarget(const char *target,
  * Otherwise:
  * Publish this message across the slot (primary/replica).
  * -------------------------------------------------------------------------- */
-void clusterPropagatePublish(robj *channel, robj *message, int sharded) {
-    clusterMsgSendBlock *msgblock;
-
+void clusterPropagatePublish(robj *channel, robj **messages, int count, int sharded) {
+    clusterMsgSendBlock *msgblock, *msgblock_light;
+    int i;
+    msgblock_light = clusterCreatePublishLightMsgBlock(
+        channel, messages, count, sharded ? CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT : CLUSTERMSG_TYPE_PUBLISH_LIGHT);
     if (!sharded) {
-        msgblock = clusterCreatePublishMsgBlock(channel, message, CLUSTERMSG_TYPE_PUBLISH);
-        clusterBroadcastMessage(msgblock);
-        clusterMsgSendBlockDecrRefCount(msgblock);
+        clusterBroadcastMessage(msgblock_light);
+        clusterMsgSendBlockDecrRefCount(msgblock_light);
+        if (!server.cluster->is_light_hdr_supported) {
+            for (i = 0; i < count; i++) {
```

Comment:
Could we assert that the count is zero? And avoid the loop for now. We can always introduce it back with `MPUBLISH` command. WDYT?